File: src/Models/Payment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DashAccountingSystemV2.Models
{
    public class Payment
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public Guid Id { get; set; }

        [Required]
        public Guid TenantId { get; set; }
        public Tenant Tenant { get; private set; }

        [Required]
        public Guid CustomerId { get; set; }
        public Customer Customer { get; private set; }

        [Required]
        public Guid DepositAccountId { get; set; }
        public Account DepositAccount { get; private set; }

        [Required]
        public Guid RevenueAccountId { get; set; }
        public Account RevenueAccount { get; private set; }

        [Required]
        public int PaymentMethodId { get; set; }
        public PaymentMethod PaymentMethod { get; private set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        public decimal Amount { get; set; }

        [Required]
        public int AssetTypeId { get; set; }
        public AssetType AssetType { get; private set; }

        [Required]
        public Guid JournalEntryId { get; set; }
        public JournalEntry JournalEntry { get; private set; }

        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime Created { get; private set; }

        [Required]
        public Guid CreatedById { get; set; }
        public ApplicationUser CreatedBy { get; private set; }

        public DateTime? Updated { get; set; }

        public Guid? UpdatedById { get; set; }
        public ApplicationUser UpdatedBy { get; private set; }

        public ICollection<InvoicePayment> Invoices { get; set; }
    }
}

File: src/ClientApp/src/models/Payment.ts
import AccountLite from './AccountLite';
import Amount from './Amount';
import CustomerLite from './Customer';
import InvoicePayment from './InvoicePayment';
import JournalEntryLite from './JournalEntryLite';
import PaymentMethod from './PaymentMethod';
import UserLite from './UserLite';

export default interface Payment {
    id?: string | null; // GUID (will be assigned by the system)
    tenantId: string | null; // GUID (required to create)
    customerId: string | null; // GUID (required to create)
    customer?: CustomerLite; // not required to create; part of response
    depositAccountId: string | null; // GUID (required to create)
    depositAccount?: AccountLite; // not required to create; part of response
    revenueAccountId: string | null; // GUID (required to create)
    revenueAccount?: AccountLite; // not required to create; part of response
    paymentMethodId: number | null; // integer ID; required to create
    paymentMethod?: PaymentMethod; // not required to create; part of response
    checkNumber: number | null; // optional unsigned integer
    paymentDate: string | null; // Date in YYYY-MM-DD format (required to create)
    amount: Amount | null; // required to create
    description: string | null; // required to create
    isPosted: boolean; // required to create but could be false
    journalEntryId?: string; // not required to create; part of response
    journalEntry?: JournalEntryLite; // not required to create; part of response
    invoices: InvoicePayment[]; // required to create
    created?: string; // timestamp in ISO 8601 format; not required to create; part of response
    createdBy?: UserLite; // not required to create; part of response
    updated?: string | null; // nullable timestamp in ISO 8601 format; not required to create; part of response
    updatedBy?: UserLite | null; // not required to create; part of response
}